docs/source/verdi/verdi_user_guide.rst

```diff
@@ -26,155 +48,108 @@ For example ``verdi process kill --help`` will display:
-However, since the type of the value is lost over the command line (as each value will be passed as a string type), the command line will have to guess the type.
-Each value will first be interpreted as an ``ID``.
-If the value cannot be mapped to the ``ID`` of an entity, it will instead be considered a partial or full ``UUID``.
-In the case where the identifier can be resolved to neither a valid ``ID`` nor a ``UUID``, the code will finally assume that the value should be interpreted as a ``LABEL``.
-In almost all cases, this approach will be able to successfully and unambiguously determine the identifier type, however, there are exceptions.
-
-Consider for example a database with the following three groups:
-
-===  =====================================  ========
-ID   UUID                                   LABEL
-===  =====================================  ========
-10   12dfb104-7b2b-4bca-adc0-1e4fd4ffcc88   group
-11   deadbeef-62ba-444f-976d-31d925dac557   10
-12   3df34a1e-5215-4e1a-b626-7f75b9586ef5   deadbeef
-===  =====================================  ========
 
-We would run into trouble if we wanted to identify the second group by its label ``10``, since it would first be interpreted as an ``ID``, which would return the first group instead.
-Likewise, if we wanted to retrieve the third group by its label, we would get the second group instead, since the label ``deadbeef`` is also a valid partial UUID of the second group.
-Finally, say we wanted to select the first group using its partial ``UUID`` ``12``, it would unfortunately match the third group on its ``ID`` instead.
+When working with AiiDA entities, you need a way to *refer* to them on the command line.
+Any entity in AiiDA can be addressed via three identifiers:
 
-Luckily, ``verdi`` provides the tools to break all of these ambiguities with guaranteed success.
-The latter ambiguity, between an ``ID`` and ``UUID`` can always be resolved by passing a larger partial ``UUID``.
-Inevitably, eventually a non-numeric character or a dash will be included in the partial ``UUID``, rendering it an invalid ``ID`` and the identifier will be cast to the right type.
-The case of an identifier, that is intended to refer to a ``LABEL``, that just happens to also be a valid ``ID`` or ``UUID`` cannot be solved in this way.
-For this case ``verdi`` reserves a special character, the exclamation mark ``!`` that can be appended to the identifier.
-Before any type guessing is done, the command line will check for the presence of this marker, and if found will directly interpret the identifier as a ``LABEL``.
-For example, to solve ambiguity problems of the first two examples given in this section, one would have had to pass ``10!`` and ``deadbeef!``.
-The exclamation point would have forced them to be interpreted as a ``LABEL`` and ensured that the right group would be retrieved.
+ * "Primary Key" (PK): An integer, e.g. ``723``, that identifies your entity within your database (automatically assigned)
```

Comment:
This is great, however, could you afterwards (or here, but preferably afterwards) even more explicitly explain that PKs are local identifiers _only_ and the exact same database content may have completely different PKs for another user. And possibly add the now obvious advantage of UUIDs :)

Reply:
thanks - I've put in a sentence on this